tensorflow/contrib/avro/python/avro_record_dataset.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import os
+
+from tensorflow.python.framework import ops
+from tensorflow.python.framework import dtypes
+from tensorflow.python.framework import load_library
+from tensorflow.python.data.ops.dataset_ops import Dataset
+from tensorflow.python.data.util import convert
+from tensorflow.python.framework import tensor_shape
+from tensorflow.python.platform import resource_loader
+from tensorflow.contrib.avro.ops.gen_avro_record_dataset import avro_record_dataset
+
+# Load the shared library
+lib_name = os.path.join(resource_loader.get_data_files_path(),
+                        '_avro_record_dataset.so')
+reader_module = load_library.load_op_library(lib_name)
+
+_DEFAULT_READER_BUFFER_SIZE_BYTES = 256 * 1024  # 256 KB
+
+
+class AvroRecordDataset(Dataset):
```

Comment:
I think this needs to be updated for TF 1.12: it should extend DatasetSource and implement `_inputs` (which can just return an empty list). There might be other parts of the code that need to be changed, too. Would suggest building against the latest TF 1.12 code if you haven't already.